x/streamer/keeper/hooks.go

```diff
@@ -0,0 +1,73 @@
+package keeper
+
+import (
+	"github.com/dymensionxyz/dymension/x/streamer/types"
+	epochstypes "github.com/osmosis-labs/osmosis/v15/x/epochs/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// BeforeEpochStart is the epoch start hook.
+func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	return nil
+}
+
+// AfterEpochEnd is the epoch end hook.
+func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	streams := k.GetUpcomingStreams(ctx)
+	// move to active if it's correct epoch and start time reached
+	for _, stream := range streams {
+		if epochIdentifier != stream.DistrEpochIdentifier {
```

Comment:
If created a stream to start at 13:05 (and distribute every hour) , and now it’s 12:55 if I understand correctly it will only start at the next time the hour epic will fire, i.e at 14? If I understand correctly we're gonna remove streams which are supposed to start based on their epoch, though their start time is not necessarily related.

Reply:
I see. not sure it's a desired behavior. IMO I'd expect a stream to start at the relevant hour defined. 
If the epoch is every week for example? could it be postponed in a few days until the week epoch will end?

Reply:
well if it's a week, it should distributed every week.
the count is not relative to the start time. it's by based on epoch triggers.

so the stream will distribute on the next epoch after the start time. make sense for me.
The only fix is that I'll change the stream to be `Active` when time come, regardless of the epoch